model_analyzer/analyzer.py

```diff
@@ -65,45 +66,64 @@ def run(self):
         TritonModelAnalyzerException
         """
 
+        config = self._config
+
         logging.info('Profiling server only metrics...')
 
+        # Phase 1: Profile server only metrics
+        self._server.start()
+        self._client.wait_for_server_ready(config.max_retries)
         self._metrics_manager.profile_server(
             default_value=SERVER_ONLY_TABLE_DEFAULT_VALUE)
+        self._server.stop()
 
-        for model in self._config.model_names:
-            self._client.load_model(model_name=model.model_name())
-            self._client.wait_for_model_ready(
-                model_name=model.model_name(),
-                num_retries=self._config.max_retries)
-            try:
-                # Send objectives and constraints to result manager
-                self._metrics_manager.configure_result_manager(
-                    config_model=model)
-
-                for run_config in FullRunConfigGenerator(
-                        analyzer_config=self._config, config_model=model):
-
-                    # Initialize the result
-                    self._result_manager.init_result(run_config)
+        output_model_repo_path = config.output_model_repository_path
+        model_repository = config.model_repository
 
-                    # TODO write/copy model configs
-                    for perf_config in run_config.perf_analyzer_configs():
-                        perf_output_writer = None if \
-                            self._config.no_perf_output else FileWriter()
-
-                        logging.info(
-                            f"Profiling model {perf_config['model-name']}...")
-                        self._metrics_manager.profile_model(
-                            perf_config=perf_config,
-                            perf_output_writer=perf_output_writer)
-
-                    # Submit the result to be sorted
-                    self._result_manager.complete_result()
-            finally:
-                self._client.unload_model(model_name=model.model_name())
-
-            # Write results to tables
-            self._result_manager.compile_results()
+        # Phase 2: Profile each model
+        for model in config.model_names:
+            self._metrics_manager.configure_result_manager(
+                    config_model=model)
+            run_config_generator = RunConfigGenerator(model,
+                                                 analyzer_config=self._config)
+            for run_config in run_config_generator.get_run_configs():
+                model_config = run_config.model_config()
+                original_model_name = run_config.model_name()
+                model_name = model_config.get_field('name')
+
+                # Create the directory for the new model
+                os.mkdir(f'{output_model_repo_path}/{model_name}')
```

Comment:
It is a good idea but I guess it is better to keep it the way it currently is for two reasons:

1. If one of the configs that the user is trying to sweep through has an issue, it can be quickly debugged by `tritonserver --model-repository <output model repository>` and find the models that cannot be loaded.
2. If we plan to integrate the multi-model analysis part, having all the models in a single folder helps quick loading of multiple models to study the co-located performance.